lib/rubocop/rspec/language.rb

```diff
@@ -2,152 +2,195 @@
+      module Helpers # :nodoc:
+        def self.all(element)
+          Language.config['Helpers'].include?(element.to_s)
+        end
       end
 
-      module Examples
-        EXAMPLES = SelectorSet.new(%i[it specify example scenario its])
-        FOCUSED  = SelectorSet.new(%i[fit fspecify fexample fscenario focus])
-        SKIPPED  = SelectorSet.new(%i[xit xspecify xexample xscenario skip])
-        PENDING  = SelectorSet.new(%i[pending])
+      module Hooks # :nodoc:
+        def self.all(element)
+          Language.config['Hooks'].include?(element.to_s)
+        end
+      end
 
-        ALL = EXAMPLES + FOCUSED + SKIPPED + PENDING
+      module HookScopes # :nodoc:
+        def self.all(element)
+          Language.config['HookScopes'].include?(element.to_s)
+        end
       end
 
-      module Hooks
-        ALL = SelectorSet.new(
-          %i[
-            prepend_before
-            before
-            append_before
-            around
-            prepend_after
-            after
-            append_after
-          ]
-        )
-
-        module Scopes
-          ALL = SelectorSet.new(
-            %i[
-              each
-              example
-              context
-              all
-              suite
-            ]
-          )
+      module Includes # :nodoc:
+        class << self
+          def all(element)
+            example(element) ||
+              context(element)
+          end
+
+          def example(element)
+            Language.config['Includes']['Example'].include?(element.to_s)
+          end
+
+          def context(element)
+            Language.config['Includes']['Context'].include?(element.to_s)
+          end
         end
       end
 
-      module Helpers
-        ALL = SelectorSet.new(%i[let let!])
+      module Runners # :nodoc:
+        def self.all(element)
+          Language.config['Runners'].include?(element.to_s)
+        end
       end
 
-      module Subject
-        ALL = SelectorSet.new(%i[subject subject!])
+      module SharedGroups # :nodoc:
+        class << self
+          def all(element)
+            example(element) ||
+              context(element)
+          end
+
+          def example(element)
+            Language.config['SharedGroups']['Example'].include?(element.to_s)
+          end
+
+          def context(element)
+            Language.config['SharedGroups']['Context'].include?(element.to_s)
+          end
+        end
       end
 
-      module Expectations
-        ALL = SelectorSet.new(%i[expect is_expected expect_any_instance_of])
+      module Subjects # :nodoc:
+        def self.all(element)
+          Language.config['Subjects'].include?(element.to_s)
+        end
       end
 
-      module Runners
-        ALL = SelectorSet.new(%i[to to_not not_to])
+      # This is used in Dialect and DescribeClass cops to detect RSpec blocks.
+      # Ommission of Expectations and Runners is a deliberate simplification,
+      # but not a requirement.
```

Comment:
Oh, I didn’t mean that they should necessarily be added back. You mentioned it was a deliberate simplification; I just wasn’t sure what that meant.

Reply:
I was on the fence with keeping (and dealing with Complexity cops) and omitting them.
Omitting them would be confusing since for our cops there's no difference, but some third-party cops could rely on ALL to match all of RSpec concepts. Seems better to have them in.